meteor/lib/mediaObjects.ts

```diff
@@ -416,23 +425,28 @@ export function checkPieceContentStatus(
 										})
 									}
 								} else if (anomalies.length > 0) {
-									const dur = anomalies.map((b) => b.duration).reduce((a, b) => a + b, 0)
+									const dur = anomalies
+										.filter((a) => sourceDuration === undefined || a.start * 1000 < sourceDuration)
+										.map((b) => b.duration)
+										.reduce((a, b) => a + b, 0)
 									const frames = Math.ceil((dur * 1000) / timebase)
-									messages.push({
-										status: PieceStatusCode.SOURCE_HAS_ISSUES,
-										message: t('{{frames}} {{type}} frames detected in the clip', {
-											frames,
-											type,
-											count: frames,
-										}),
-									})
+									if (frames > 0) {
+										messages.push({
+											status: PieceStatusCode.SOURCE_HAS_ISSUES,
+											message: t('{{frames}} {{type}} frames detected in the clip', {
+												frames,
+												type,
+												count: frames,
+											}),
+										})
+									}
 								}
 							}
-							if (deepScan?.blacks) {
-								addFrameWarning(deepScan.blacks, t('black'), t)
```

Comment:
removing `t(` here will cause translations to not be generated for this string anymore, which is probably not desired?